src/core/execute.c

```diff
@@ -948,8 +949,13 @@ static int setup_pam(
                  * and this will make PR_SET_PDEATHSIG work in most cases.
                  * If this fails, ignore the error - but expect sd-pam threads
                  * to fail to exit normally */
+
+                if (setgroups(0, NULL) < 0)
+                        log_warning_errno(errno, "Failed to setgroups() in sd-pam: %m");
+                if (setresgid(gid, gid, gid) < 0)
+                        log_warning_errno(errno, "Failed to setresgid() in sd-pam: %m");
```

Comment:
what do you mean? I think it's totally fine if the auxiliary group list is empty in this case. We don't really care for the full cred initialiuation here. We only drop privs because we can here, because there's no need to stay root in the PAM stub.

Note that we don't even check whether the dropping of privs really worked, because it doesn't matter too much: if we remain privileged here, it's not a complete loss...